pydomcfg/tests/bathymetry.py

```diff
@@ -69,17 +117,53 @@ def _add_attributes(ds: Dataset) -> Dataset:
     Parameters
     ----------
     ds: Dataset
-        Dataset with bathymetry and mask variables
+        Dataset with bathymetry and mask variables [and rmax]
 
     Returns
     -------
     Dataset
     """
+
     attrs_dict = {
         "Bathymetry": dict(standard_name="sea_floor_depth_below_geoid", units="m"),
         "mask": dict(standard_name="sea_binary_mask", units="1"),
     }
+
+    if "rmax" in list(ds.keys()):
+        attrs_dict["rmax"] = dict(standard_name="rmax", units="1")
+
     for varname, attrs in attrs_dict.items():
         ds[varname].attrs = attrs
         ds[varname].attrs["coordinates"] = "glamt gphit"
     return ds
+
+
+def _calc_rmax(depth):
+    """
+    Calculate rmax: measure of steepness
+
+    This function returns the slope steepness criteria rmax, which is simply
+    (H[0] - H[1]) / (H[0] + H[1])
+
+    Parameters
+    ----------
+    depth: float
+            Bottom depth (units: m).
+
+    Returns
+    -------
+    rmax: float
+            Slope steepness value (units: None)
+    """
+    rmax_x, rmax_y = np.zeros_like(depth), np.zeros_like(depth)
+
+    rmax_x[:, 1:-1] = 0.5 * (
+        np.diff(depth[:, :-1], axis=1) / (depth[:, :-2] + depth[:, 1:-1])
+        + np.diff(depth[:, 1:], axis=1) / (depth[:, 1:-1] + depth[:, 2:])
+    )
+    rmax_y[1:-1, :] = 0.5 * (
+        np.diff(depth[:-1, :], axis=0) / (depth[:-2, :] + depth[1:-1, :])
+        + np.diff(depth[1:, :], axis=0) / (depth[1:-1, :] + depth[2:, :])
+    )
+
+    return np.maximum(np.abs(rmax_x), np.abs(rmax_y))
```

Comment:
This is how I would do this:
```python
    # Let's forget about indexes here,
    # will be restored when we merge in the dataset
    depth = depth.reset_index(list(depth.dims))
    
    both_rmax = []
    for dim in depth.dims:
        
        # (H[0] - H[1]) / (H[0] + H[1])
        depth_diff = depth.diff(dim)
        depth_rolling_sum = depth.rolling({dim: 2}).sum().dropna(dim)
        rmax = depth_diff / depth_rolling_sum
        
        # (R[0] + R[1]) / 2
        rmax = rmax.rolling({dim: 2}).mean().dropna(dim)
        
        # Fill boundaries
        rmax = rmax.pad({dim: (1, 1)}, constant_values=0)  
        
        both_rmax.append(np.abs(rmax))

    return np.maximum(*both_rmax)
```
Both input and output is a DataArray, so when you call it you can just do this:
```python
ds["rmax"] = _calc_rmax(ds["Bathymetry"])
```

Reply:
Yes Master ;) Haha ... I had written a note to myself to ask you about this tomorrow. I had tried passing the DataArray when I first wrote the function, but kept getting broadcast errors (probably because I was mixing np with xr!) ... I'll take a moment after lunch to digest your tutorial!

Reply:
eheheh. Just wanted to show this rolling stuff before tomorrow. In general most of `np` functions should work OK. The problem is probably that there are many diff/rolling operations, and you need to make sure that all indexes are in good shape.
Or, if you don't want to do that, you just drop them like I did and they're always set to `range(len(dim))`